packages/mobx-state-tree/src/core/node/identifier-cache.ts

```diff
@@ -68,23 +68,31 @@ export class IdentifierCache {
         }
     }
 
-    splitCache(node: AnyObjectNode): IdentifierCache {
-        const res = new IdentifierCache()
-        const basePath = node.path
+    splitCache(splitNode: AnyObjectNode): IdentifierCache {
+        const newCache = new IdentifierCache()
+        // The slash is added here so we only match children of the splitNode. In version 5.1.8 and
+        // earlier there was no trailing slash, so non children that started with the same path string
+        // were being matched incorrectly.
+        const basePath = splitNode.path + "/"
         entries(this.cache).forEach(([id, nodes]) => {
             let modified = false
             for (let i = nodes.length - 1; i >= 0; i--) {
-                if (nodes[i].path.indexOf(basePath) === 0) {
-                    res.addNodeToCache(nodes[i], false) // no need to update lastUpdated since it is a whole new cache
+                const node = nodes[i]
+                if (node === splitNode || node.path.indexOf(basePath) === 0) {
+                    newCache.addNodeToCache(node, false) // no need to update lastUpdated since it is a whole new cache
                     nodes.splice(i, 1)
+                    // remove empty sets from cache
+                    if (!nodes.length) {
+                        this.cache.delete(id)
+                    }
                     modified = true
                 }
             }
             if (modified) {
                 this.updateLastCacheModificationPerId(id)
             }
         })
-        return res
+        return newCache
```

Comment:
**praise:** @scytacki - I like this name a little more than `res` here. Nice improvement!